### PR TITLE
Fee credits: automatically return to organisations

### PR DESCRIPTION
**Pull request by sampoder**
## Summary of the problem

It's weird! Some orgs have had fee credits for years.

## Describe your changes

Now we automatically refund people, see https://hackclub.slack.com/archives/C047Y01MHJQ/p1706558034009749 for context on implementation

### CODE REVIEW

**Comment by sampoder** [comment]
Closes https://github.com/hackclub/hcb/issues/5133

File: app/services/bank_fee_service/weekly.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ def run
 
       FeeRevenue.create!(
         bank_fees:,
-        amount_cents: bank_fees.sum { |fee| fee.amount_cents.abs },
+        amount_cents: bank_fees.sum { |fee| fee.amount_cents * -1 },
```
Comment (cjdenio) [inline app/services/bank_fee_service/weekly.rb]:
```suggestion
        amount_cents: bank_fees.sum { |fee| -fee.amount_cents },
```

**Comment by sampoder** [comment]
A couple things: the memo right now would be `FISCAL SPONSORSHIP`, do we want to change that?

**Comment by sampoder** [comment]
And in the rare case (which is now possible) that `FeeRevenue` is actually a negative amount would we want to make the memo:

```
"Fee revenue from #{fee_revenue.start.strftime("%b %e")} to #{fee_revenue.end.strftime("%b %e")}"
```

something different?

**Comment by garyhtou** [comment]
> And in the rare case (which is now possible) that `FeeRevenue` is actually a negative amount would we want to make the memo:
> 
> ```
> "Fee revenue from #{fee_revenue.start.strftime("%b %e")} to #{fee_revenue.end.strftime("%b %e")}"
> ```
> 
> something different?

that's okay, we can leave it as is

**Comment by garyhtou** [comment]
> A couple things: the memo right now would be `FISCAL SPONSORSHIP`, do we want to change that?

let's change this to "FISCAL SPONSORSHIP FEE CREDIT"

File: app/services/bank_fee_service/create.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ def initialize(event_id:)
 
     def run
       raise ArgumentError, "must be an event that has not had a fee for more than 5 days" unless event.ready_for_fee?
-      raise ArgumentError, "must be an event that has a balance greater than 0" unless event.fee_balance_v2_cents > 0
+      raise ArgumentError, "must be an event that has a non-zero balance" unless event.fee_balance_v2_cents != 0
```
Comment (garyhtou) [inline app/services/bank_fee_service/create.rb]:
```suggestion
      raise ArgumentError, "must be an event that has a non-zero fee balance" if event.fee_balance_v2_cents.zero?
```

File: app/models/hcb_code/memo.rb [hunk 3]
```diff
@@ -60,7 +60,7 @@ def partner_donation_memo
       end
 
       def bank_fee_memo
-        "FISCAL SPONSORSHIP"
+        bank_fee.amount_cents < 0 ? "FISCAL SPONSORSHIP" : "FISCAL SPONSORSHIP FEE CREDIT"
```
Comment (garyhtou) [inline app/models/hcb_code/memo.rb]:
```suggestion
        bank_fee.amount_cents.negative? ? "FISCAL SPONSORSHIP" : "FISCAL SPONSORSHIP FEE CREDIT"
```

File: app/services/bank_fee_service/process_single.rb [hunk 4]
```diff
@@ -12,11 +12,15 @@ def run
       ActiveRecord::Base.transaction do
         bank_fee.mark_in_transit!
 
+        sender_bank_account_id = amount_cents < 0 ? ColumnService::Accounts::FS_MAIN : ColumnService::Accounts::FS_OPERATING
+
+        receiver_bank_account_id = amount_cents < 0 ? ColumnService::Accounts::FS_OPERATING : ColumnService::Accounts::FS_MAIN
```
Comment (garyhtou) [inline app/services/bank_fee_service/process_single.rb]:
```suggestion
        sender_bank_account_id = amount_cents.negative? ? ColumnService::Accounts::FS_MAIN : ColumnService::Accounts::FS_OPERATING

        receiver_bank_account_id = amount_cents.negative? ? ColumnService::Accounts::FS_OPERATING : ColumnService::Accounts::FS_MAIN
```

File: app/services/fee_revenue_service/process_single.rb [hunk 5]
```diff
@@ -12,11 +12,15 @@ def run
       ActiveRecord::Base.transaction do
         fee_revenue.mark_in_transit!
 
+        sender_bank_account_id = amount_cents < 0 ? ColumnService::Accounts::FS_OPERATING : ColumnService::Accounts::FS_MAIN
+
+        receiver_bank_account_id = amount_cents < 0 ? ColumnService::Accounts::FS_MAIN : ColumnService::Accounts::FS_OPERATING
```
Comment (garyhtou) [inline app/services/fee_revenue_service/process_single.rb]:
```suggestion
        sender_bank_account_id = amount_cents.negative? ? ColumnService::Accounts::FS_OPERATING : ColumnService::Accounts::FS_MAIN

        receiver_bank_account_id = amount_cents.negative? ? ColumnService::Accounts::FS_MAIN : ColumnService::Accounts::FS_OPERATING
```

**Review by garyhtou — APPROVED** [review]

**Pull request merged**